src/internal_modules/roc_core/parse_units.cpp

```diff
@@ -84,5 +84,51 @@ bool parse_duration(const char* str, nanoseconds_t& result) {
     return true;
 }
 
+bool parse_size(const char* str, size_t& result) {
+    if (str == NULL) {
+        roc_log(LogError, "parse size: string is null");
+        return false;
+    }
+
+    const size_t kibibyte = 1024;
+    const size_t mebibyte = 1024 * kibibyte;
+    const size_t gibibyte = 1024 * mebibyte;
+
+    size_t multiplier = 1;
+
+    const size_t str_len = strlen(str);
+    const char* suffix;
+
+    // suffix is optional.
+    if ((suffix = find_suffix(str, str_len, "G"))) {
+        multiplier = gibibyte;
+    } else if ((suffix = find_suffix(str, str_len, "M"))) {
+        multiplier = mebibyte;
+    } else if ((suffix = find_suffix(str, str_len, "K"))) {
+        multiplier = kibibyte;
+    }
+
+    if (!isdigit(*str)) {
+        roc_log(LogError,
+                "parse size: invalid format, not a number, expected <number>[<suffix>], "
+                "where suffix=<K|M|G>");
+        return false;
+    }
+
+    char* number_end = NULL;
+    unsigned long long number = strtoull(str, &number_end, 10);
+
+    if (number == ULLONG_MAX || (!suffix && *number_end != '\0')
+        || (suffix && number_end != suffix)) {
+        roc_log(LogError,
+                "parse size: invalid format, can't parse number, expected "
+                "<number>[<suffix>], where suffix=<K|M|G>");
+        return false;
+    }
+
+    result = number * multiplier;
+    return true;
```

Comment:
We should also check for two possible overflows:

* if size_t is 32-bit, and long long is 64-bit, number may not fit size_t
* even if size_t is 64-bit, `number * multiplier` may not fit size_t

Reply:
Also would be nice to cover those overflows in tests.